src/RA_Core.cpp

```diff
@@ -734,52 +748,52 @@ API int CCONV _RA_HandleHTTPResults()
-        AppendMenu(hRA, MF_STRING, IDM_RA_GETROMCHECKSUM, TEXT("Get ROM &Checksum"));
-        AppendMenu(hRA, MF_STRING, IDM_RA_SCANFORGAMES, TEXT("Scan &for games"));
+        AppendMenu(hRA.get(), nGameFlags, IDM_RA_OPENGAMEPAGE, TEXT("Open this &Game's Page"));
+        AppendMenu(hRA.get(), MF_SEPARATOR, 0U, nullptr);
+        AppendMenu(hRA.get(), pConfiguration.IsFeatureEnabled(ra::services::Feature::Hardcore) ? MF_CHECKED : MF_UNCHECKED, IDM_RA_HARDCORE_MODE, TEXT("&Hardcore Mode"));
+        AppendMenu(hRA.get(), MF_SEPARATOR, 0U, nullptr);
+
+        AppendMenu(hRA.get(), MF_POPUP, reinterpret_cast<UINT_PTR>(hRA_LB.get()), TEXT("Leaderboards"));
+        AppendMenu(hRA_LB.get(), pConfiguration.IsFeatureEnabled(ra::services::Feature::Leaderboards) ? MF_CHECKED : MF_UNCHECKED, IDM_RA_TOGGLELEADERBOARDS, TEXT("Enable &Leaderboards"));
+        AppendMenu(hRA_LB.get(), MF_SEPARATOR, 0U, nullptr);
+        AppendMenu(hRA_LB.get(), pConfiguration.IsFeatureEnabled(ra::services::Feature::LeaderboardNotifications) ? MF_CHECKED : MF_UNCHECKED, IDM_RA_TOGGLE_LB_NOTIFICATIONS, TEXT("Display Challenge Notification"));
+        AppendMenu(hRA_LB.get(), pConfiguration.IsFeatureEnabled(ra::services::Feature::LeaderboardCounters) ? MF_CHECKED : MF_UNCHECKED, IDM_RA_TOGGLE_LB_COUNTER, TEXT("Display Time/Score Counter"));
+        AppendMenu(hRA_LB.get(), pConfiguration.IsFeatureEnabled(ra::services::Feature::LeaderboardScoreboards) ? MF_CHECKED : MF_UNCHECKED, IDM_RA_TOGGLE_LB_SCOREBOARD, TEXT("Display Rank Scoreboard"));
+
+        AppendMenu(hRA.get(), MF_SEPARATOR, 0U, nullptr);
+        AppendMenu(hRA.get(), MF_STRING, IDM_RA_FILES_ACHIEVEMENTS, TEXT("Achievement &Sets"));
+        AppendMenu(hRA.get(), MF_STRING, IDM_RA_FILES_ACHIEVEMENTEDITOR, TEXT("Achievement &Editor"));
+        AppendMenu(hRA.get(), MF_STRING, IDM_RA_FILES_MEMORYFINDER, TEXT("&Memory Inspector"));
+        AppendMenu(hRA.get(), MF_STRING, IDM_RA_PARSERICHPRESENCE, TEXT("Rich &Presence Monitor"));
+        AppendMenu(hRA.get(), MF_SEPARATOR, 0U, nullptr);
+        AppendMenu(hRA.get(), MF_STRING, IDM_RA_REPORTBROKENACHIEVEMENTS, TEXT("&Report Broken Achievements"));
+        AppendMenu(hRA.get(), MF_STRING, IDM_RA_GETROMCHECKSUM, TEXT("Get ROM &Checksum"));
+        AppendMenu(hRA.get(), MF_STRING, IDM_RA_SCANFORGAMES, TEXT("Scan &for games"));
     }
     else
     {
-        AppendMenu(hRA, MF_STRING, IDM_RA_FILES_LOGIN, TEXT("&Login to RA"));
+        AppendMenu(hRA.get(), MF_STRING, IDM_RA_FILES_LOGIN, TEXT("&Login to RA"));
     }
 
-    AppendMenu(hRA, MF_SEPARATOR, 0U, nullptr);
-    AppendMenu(hRA, MF_STRING, IDM_RA_FILES_CHECKFORUPDATE, TEXT("&Check for Emulator Update"));
+    AppendMenu(hRA.get(), MF_SEPARATOR, 0U, nullptr);
+    AppendMenu(hRA.get(), MF_STRING, IDM_RA_FILES_CHECKFORUPDATE, TEXT("&Check for Emulator Update"));
 
-    return hRA;
+    return hRA.release(); // must be released
```

Comment:
If we don't release it, the menu will not appear. Behavior-wise, this is exactly how it was originally except the other handle is deallocated. But the leaderboard menu still appears since `hRA `owns `hRA_LB`. This just makes it clear that `hRA` is not owned anymore. If you think this causes a leak, then what was here before also causes a leak.